interactive-Style.js

```
@@ -0,0 +1,115 @@
+import { css } from 'lit-element';
+
+export const interactiveStyle = css`
+
+  :host([disabled]) {
+    pointer-events: none;
+    color: var(--mdc-theme-text-disabled-on-surface);
+  }
+
+  :host([disabled][bg="primary"]),
+  :host([disabled][bg="secondary"]),
+  :host([disabled][bg="error"]) {
+    background-color: var(--mdc-theme-surface);
+  }
+
+  /* STARTS: overlay styles for hover state */
+  :host(:not([disabled])[interactive]) .overlay {
+    pointer-events: auto;
+  }
+
+  :host([interactive]:hover) .overlay {
+    background-color: rgba(0, 0, 0, 0.04);
```

instead of hard-coding the color, we should use theme colors for "on-surface", "on-primary", "on-secondary" and "on-error". As a fallback we may use the constant value though. This change is required everywhere in this implementation